.github/workflows/typescript_test.yml

```diff
@@ -28,90 +28,65 @@ on:
 env:
   NODE_VERSION: "21"
   PYTHON_VERSION: "3.12"
-  # Define the directory where Playwright browsers will be installed.
-  # Adjust if your project uses a different path.
   PLAYWRIGHT_BROWSERS_PATH: "ms-playwright"
 
 jobs:
+  determine-test-suite:
+    runs-on: ubuntu-latest
+    outputs:
+      suites: ${{ steps.set-suites.outputs.suites }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref || github.ref }}
+          fetch-depth: 0
+
+      - id: set-suites
+        run: |
+          SUITES="[]"
+          CHANGES=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }})
+
+          if echo "$CHANGES" | grep -q "src/backend/base/langflow/initial_setup/\|src/backend/base/langflow/\|src/frontend/workspace/"; then
+            SUITES=$(echo $SUITES | jq '. += ["starter-projects"]')
+          fi
+          if echo "$CHANGES" | grep -q "src/frontend/src/components/"; then
+            SUITES=$(echo $SUITES | jq '. += ["components"]')
+          fi
+          if echo "$CHANGES" | grep -q "src/frontend/workspace/"; then
+            SUITES=$(echo $SUITES | jq '. += ["workspace"]')
+          fi
+          if echo "$CHANGES" | grep -q "src/frontend/api/"; then
+            SUITES=$(echo $SUITES | jq '. += ["api"]')
+          fi
+          if echo "$CHANGES" | grep -q "src/frontend/database/"; then
+            SUITES=$(echo $SUITES | jq '. += ["database"]')
+          fi
+
+          # If no specific changes detected, run all suites
+          if [ "$SUITES" == "[]" ]; then
+            SUITES='["starter-projects", "components", "workspace", "api", "database"]'
+          fi
+
+          echo "suites=$SUITES" >> $GITHUB_OUTPUT
```

Comment:
Maybe something similar to this could be better. Possibly using the changes-filter file we have.
```yaml

      - name: Paths Filter
        id: filter
        uses: dorny/paths-filter@v2
        with:
          filters: |
            starter-projects:
              - 'src/backend/base/langflow/initial_setup/**'
              - 'src/backend/base/langflow/**'
              - 'src/frontend/workspace/**'
            components:
              - 'src/frontend/src/components/**'
            workspace:
              - 'src/frontend/workspace/**'
            api:
              - 'src/frontend/api/**'
            database:
              - 'src/frontend/database/**'
          base: ${{ github.event.before }}
          head: ${{ github.sha }}

      - name: Set Matrix
        id: set-matrix
        run: |
          suites=()
          if [ "${{ steps.filter.outputs.starter-projects_changed }}" == "true" ]; then
            suites+=("starter-projects")
          fi
          if [ "${{ steps.filter.outputs.components_changed }}" == "true" ]; then
            suites+=("components")
          fi
          if [ "${{ steps.filter.outputs.workspace_changed }}" == "true" ]; then
            suites+=("workspace")
          fi
          if [ "${{ steps.filter.outputs.api_changed }}" == "true" ]; then
            suites+=("api")
          fi
          if [ "${{ steps.filter.outputs.database_changed }}" == "true" ]; then
            suites+=("database")
          fi
          if [ ${#suites[@]} -eq 0 ]; then
            suites=("starter-projects" "components" "workspace" "api" "database")
          fi
          matrix=$(printf '{"suite":%s}' "$(printf '%s\n' "${suites[@]}" | jq -R . | jq -s .)")
          echo "matrix=$matrix" >> $GITHUB_OUTPUT
  ```